command/agent/http.go

```diff
@@ -277,6 +277,11 @@ func handleUI(h http.Handler) http.Handler {
 
 func handleRootRedirect() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/" {
```

Comment:
I think we'll want all `/ui/*` routes to redirect to `/ui/` too. Otherwise navigating to a UI deep link will 404.

The easiest way to test this is to run the UI via `make dev-ui`, go to a job or something, and try refreshing the page.

Reply:
Let me double check but that should be covered here https://github.com/hashicorp/nomad/blob/master/command/agent/http.go#L208-L209

Reply:
hard refreshes on something like http://localhost:4646/ui/jobs/example stay where they are supposed to

Reply:
Oh neat. Disregard then.